### PR TITLE
YAML-indicators with custom indices + doc

### PR DESCRIPTION
**Pull request by aulemahal**
### Pull Request Checklist:
- [ ] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #xyz
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
- [ ] `bumpversion (major / minor / patch)` has been called on this branch
- [ ] Tags have been pushed (`git push --tags`)

* **What kind of change does this PR introduce?** 

   - A mapping (dict) or module can now be passed to `build_indicator_modules_from_yaml` through it's `indices` keyword. Then, the name listed in the `index_function.name` YAML field is first sought in this mapping, then in `xclim.indices.generic` and finally in `xclim.indices`.
   - An `iter_indicators()` function is injected in virtual submodules. It iterates over pairs of (name, indicator obj), making it cleaner to use those modules as "collections".
   - Update the "Extend xclim" notebook with visuals from my presentation to Ouranos.
   - Accept period specification in the YAML with : 
   
```yaml
period: annual
# or
period:
  season: [JFM, AMJJAS, OND]
```
instead of
```yaml
period:
    allowed:
        annual:
    default: annual
```
As discussed with the clix-meta gang (clix-meta/clix-meta#29). However, xclim doesn't support the season specification : in the second example above, the function will default to "QS-DEC", the "JFM" part is not parsed.

* **Does this PR introduce a breaking change?** 
No. The signature of `Indicator.from_dict` changed, but I doubt anyone was using it directly.

* **Other information**:

### CODE REVIEW

**Review by huard — APPROVED** [review]
Is there a test for this new functionality? I don't understand enough what's going on to guess correctly.

File: docs/notebooks/extendxclim.ipynb [hunk 1]
```diff
@@ -8,29 +8,43 @@
     "\n",
```
Comment (huard) [inline docs/notebooks/extendxclim.ipynb:8]: <p>extra "(e" at the end. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Ouranosinc/xclim/pull/725/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (huard) [inline docs/notebooks/extendxclim.ipynb:8]: <p>f</p><p>rom</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Ouranosinc/xclim/pull/725/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

**Comment by aulemahal** [comment]
The test is the notebook. But I just found some other errors and will write a real test.

The real new functionality is all that is added in `indicators.py`:  lines ~478, then 1138 and below. It is the one I use for the `R95p` indicator in the example.

**Pull request merged**